server/etcdserver/api/membership/cluster.go

```diff
@@ -339,14 +342,14 @@ func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange) error {
 				}
 			}
 
-			if confChangeContext.Member.IsLearner { // the new member is a learner
+			if confChangeContext.Member.RaftAttributes.IsLearner && cc.Type == raftpb.ConfChangeAddLearnerNode { // the new member is a learner
 				numLearners := 0
 				for _, m := range members {
 					if m.IsLearner {
 						numLearners++
 					}
 				}
-				if numLearners+1 > maxLearners {
+				if numLearners+1 > c.maxLearners {
```

Comment:
Should we rather call ValidateMaxLearnerConfig here (replacing LOC 346 to 353) ?

Reply:
Yeah, this is N+1 scale up but I would like to generalize the function to accommodate. Will resolve.

Reply:
Thanks @hexfusion